pandas-stubs/core/arrays/datetimes.pyi

```diff
@@ -8,18 +9,38 @@ from pandas.core.arrays.datetimelike import (
     DatetimeLikeArrayMixin,
     TimelikeOps,
 )
+from pandas.core.arrays.period import PeriodArray
+from pandas.core.indexes.base import Index
+from pandas.core.series import Series
+from typing_extensions import (
+    Never,
+    Self,
+)
 
+from pandas._libs.tslibs.timestamps import Timestamp
 from pandas._typing import (
+    DtypeArg,
+    Frequency,
     TimeAmbiguous,
     TimeNonexistent,
     TimeZones,
+    np_1darray,
+    np_ndarray,
+    np_ndarray_float,
+    np_ndarray_str,
 )
 
 from pandas.core.dtypes.dtypes import DatetimeTZDtype as DatetimeTZDtype
 
 class DatetimeArray(DatetimeLikeArrayMixin, TimelikeOps, DatelikeOps):
     __array_priority__: int = ...
-    def __init__(self, values, dtype=..., freq=..., copy: bool = ...) -> None: ...
+    def __init__(
+        self,
+        values: Series | Index | DatetimeArray | np_ndarray_str,
```

Comment:
numpy array of type datetime64 is also allowed. wouldn't be averse to just np_ndarray, as `Series` and `Index` are also non-specialised here?